File: core/src/main/java/co/ericp/freewayforecast/routes/Route.java
package co.ericp.freewayforecast.routes;

import java.util.List;

import co.ericp.freewayforecast.LatLng;
import co.ericp.freewayforecast.Location;

/**
 * A route at a particular time from a place to a place.
 */
public class Route {
    double distance; // distance in meters
    long duration;
    long startTime;
    long endTime;
    LatLng neBound;
    LatLng swBound;
    Location origin;
    Location destination;

    List<Leg> legs;
}

File: core/src/main/java/co/ericp/freewayforecast/routes/Route.kt
package co.ericp.freewayforecast.routes

import co.ericp.freewayforecast.Location

/**
 * A route at a particular time from a place to a place.
 */
class Route(
        val summary: String,
        val distance: Double, // Meters
        val duration: Long, // Milliseconds
        val startTime: Long, // Unix time (millis)
        val endTime: Long,
        val neBound: Location,
        val swBound: Location,
        val origin: Location,
        val destination: Location,
        val legs: List<Leg>
)
